src/web_interface/file_tree/file_tree.py

```diff
@@ -120,15 +76,21 @@ def root_is_virtual(root: List[dict]) -> bool:
 
 def remove_virtual_path_from_root(root: List[dict]) -> List[dict]:
     '''
-    when a file object is the root, the directories that contain the file object need to be removed so that the file
-    tree is displayed correctly in the web interface
+    When a file object is the root, the directories that contain the file object need to be removed so that the file
+    tree is displayed correctly in the web interface.
     '''
-    while root_is_virtual(root):
+    while _root_is_virtual(root):
         root = root[0]['children']
     return root
 
 
 class VirtualPathFileTree:
+    '''
+    This class represents a layer or the  file tree  with a ``Firmware`` or ``FileObject`` as root, directories as
```

Comment:
I don't understand what that's suppose to mean.

Reply:
"Layer" means that the file tree is created in layers as it is unfolded (one partial tree for each file)
This partial layer tree has a firmware or FO as root, directories as inner nodes (files can't be inner nodes) and FOs as outer nodes ("leaves" of the tree)
Note that the leaves of this tree may not be actual leaves in the full file tree. Instead, a partial ("layer") tree is created for each file (for efficiency)

Reply:
updated the docstring

Reply:
See. Now I get it.